Add orm and typescript

andriy-shymkiv opened the pull request:
- add typescript for the whole service and ORM for database connection

andriy-shymkiv committed:
add prettier config

andriy-shymkiv committed:
prettier format

andriy-shymkiv committed:
add prettier package

andriy-shymkiv committed:
add socket.io

andriy-shymkiv committed:
configure basic io connection

andriy-shymkiv committed:
add transactions event emitter

andriy-shymkiv committed:
change limit

andriy-shymkiv committed:
rename from transactions to transactions-history

andriy-shymkiv committed:
encapsulate transaction-history event

andriy-shymkiv committed:
update event constant

andriy-shymkiv committed:
refactor transactions history args and structure

andriy-shymkiv committed:
update readme and env example

andriy-shymkiv committed:
add state to store interval and last tx block_num & refactoring

andriy-shymkiv committed:
refactor transactions history functions

andriy-shymkiv committed:
state handlers, event_logs scanning and refactoring

andriy-shymkiv committed:
switch from trace to fork transactions if irreversible is true & adjustments

andriy-shymkiv committed:
update queries

andriy-shymkiv committed:
update readme and socket

andriy-shymkiv committed:
decompose emitting functions

andriy-shymkiv committed:
allow to connect only once per IP

andriy-shymkiv committed:
refactor emitTrace and emitFork functions, add comments and getHeadBlockQuery function

andriy-shymkiv committed:
Refactor emitTransactionsHistory function to handle different transaction types

andriy-shymkiv committed:
refactor WebSocket transactions history scanning and emitting

andriy-shymkiv committed:
update readme

andriy-shymkiv committed:
update readme

andriy-shymkiv committed:
update comments

andriy-shymkiv committed:
add typescript

andriy-shymkiv committed:
add more types

andriy-shymkiv committed:
remove logs

andriy-shymkiv committed:
remove ip address limitation

andriy-shymkiv committed:
fix transactionType assignment and add null checks

andriy-shymkiv committed:
implement new requirements

andriy-shymkiv committed:
update WebSocket transaction scanning and add new configuration options

andriy-shymkiv committed:
add import to readme

andriy-shymkiv committed:
update fork's state

andriy-shymkiv committed:
add jest, update tsconfig, update websocket service

andriy-shymkiv committed:
change back config

andriy-shymkiv committed:
add path

andriy-shymkiv committed:
add socket 'transports' option

andriy-shymkiv committed:
update systemd

andriy-shymkiv committed:
update readme

andriy-shymkiv committed:
update error handling

andriy-shymkiv committed:
update query

andriy-shymkiv committed:
update event_log writing

andriy-shymkiv committed:
refactor WebSocket transactions history logic

andriy-shymkiv committed:
refactor emitTransactionsHistory function and add scheduleNextEmit function

andriy-shymkiv committed:
remove jest config & packages

andriy-shymkiv committed:
removed todo and add ack callback to readme

andriy-shymkiv committed:
rename func

andriy-shymkiv committed:
rename func back

andriy-shymkiv committed:
update readme

andriy-shymkiv committed:
update socket path

andriy-shymkiv committed:
add code editor to ignore

andriy-shymkiv committed:
add sequelize setup

andriy-shymkiv committed:
add models

andriy-shymkiv committed:
update database configuration

andriy-shymkiv committed:
add services for each entity and simplify the websocket service

andriy-shymkiv committed:
add ts to routes

andriy-shymkiv committed:
add ts to health module

andriy-shymkiv committed:
add time depend on database

andriy-shymkiv committed:
add TS to transactionStatus module

andriy-shymkiv committed:
show the exact reason and a place where the error  occurred during runtime

andriy-shymkiv committed:
Refactor TransactionStatus controller

andriy-shymkiv committed:
update and add typing

andriy-shymkiv committed:
change order of res.data

andriy-shymkiv committed:
Refactor TransactionStatus controller to include block_time and block_num in ReadTransactionOutput

andriy-shymkiv committed:
add ts to graphql router

andriy-shymkiv committed:
rename graphql controller to .ts

andriy-shymkiv committed:
add ts to graphql controller

andriy-shymkiv committed:
remove unused package

andriy-shymkiv committed:
rename func, remove unused import

andriy-shymkiv committed:
change history route to .ts

andriy-shymkiv committed:
add ts to history controller

andriy-shymkiv committed:
change history controller to .ts

andriy-shymkiv committed:
add type for trace and todo to define it

andriy-shymkiv committed:
adding typescript in progress

andriy-shymkiv committed:
update unhandledRejection

cc32d9 commented:
Please add logging: on every websocket request, print the details of the request. Also, print errors if there are any.

andriy-shymkiv committed:
add some logging

andriy-shymkiv committed:
add another logs and updates

andriy-shymkiv committed:
update readme

Commit:
update

Commit:
add another log

MaksHladun committed:
Fix typo in WebSocket event name

MaksHladun committed:
Update WebSocket event listener to include accounts in transaction history

MaksHladun committed:
Update WebSocket event listener to include accounts in transaction history

MaksHladun committed:
Refactor WebSocket event listener to include headBlock in transaction history

MaksHladun committed:
Refactor WebSocket event listener to include startBlock in transaction history

MaksHladun committed:
Refactor WebSocket event listener to remove unnecessary code and improve readability

MaksHladun committed:
Refactor WebSocket event listener to rename transactionType to eventType and update related types

MaksHladun committed:
Refactor WebSocket event listener to fix typo in emitTransactionHistory function

MaksHladun committed:
Fix typo in emitTransactionBasedOnType function

MaksHladun committed:
Refactor WebSocket event listener to use transactionService instead of transactionsService in transactionHistory/index.ts

MaksHladun committed:
Add getNextBlockWithTransaction function to receipts service

MaksHladun committed:
Refactor WebSocket event listener to fix typo and improve transaction history in transactionHistory/index.ts

MaksHladun committed:
Refactor WebSocket event listener to update event type handling in transactionHistory/index.ts

MaksHladun committed:
Refactor getNextBlockWithTransaction function in receipts service

MaksHladun committed:
Refactor WebSocket event listener to use manageEventLogSaveInState instead of manageForkEventSaveInState in transactionHistory/index.ts

MaksHladun committed:
Refactor WebSocket event listener to update event type handling and improve transaction history in transactionHistory/utils.ts

MaksHladun committed:
Refactor WebSocket event listener to update event type handling and improve transaction history in transactionHistory/types.ts

MaksHladun committed:
Refactor WebSocket event listener to update event type handling and improve transaction history in transactionHistory/types.ts
Refactor WebSocket event listener to update event type handling and improve transaction history in transactionHistory/utils.ts
Refactor WebSocket event listener to use manageEventLogSaveInState instead of manageForkEventSaveInState in transactionHistory/index.ts
Refactor getNextBlockWithTransaction function in receipts service
Refactor WebSocket event listener to update event type handling in transactionHistory/index.ts
Refactor WebSocket event listener to fix typo and improve transaction history in transactionHistory/index.ts
Add getNextBlockWithTransaction function to receipts service
Refactor WebSocket event listener to use transactionService instead of transactionsService in transactionHistory/index.ts
Fix typo in emitTransactionBasedOnType function
Refactor WebSocket event listener to fix typo in emitTransactionHistory function

MaksHladun committed:
Update WebSocket event listener to use WS_EVENTLOG_TRANSACTIONS_LIMIT instead of WS_FORK_TRANSACTIONS_LIMIT

MaksHladun committed:
Update WebSocket event listener to use WS_EVENTLOG_TRANSACTIONS_LIMIT instead of WS_FORK_TRANSACTIONS_LIMIT

MaksHladun committed:
Refactor WebSocket event listener to filter and format event logs in webSocketFormat function

MaksHladun committed:
Refactor WebSocket event listener to include console.log statement for testing purposes

MaksHladun committed:
Refactor eventLog and transactionHistory services to improve logging and debugging

MaksHladun committed:
Refactor WebSocket event listener to update event type handling and improve transaction history in transactionHistory/index.ts

MaksHladun committed:
Refactor WebSocket event listener to include console.log statement for testing purposes

MaksHladun committed:
Refactor logging and debugging in eventLog and transactionHistory services

MaksHladun committed:
Refactor server.ts to validate and handle numeric environment options
